aws/cloudformation/iam.yml.erb

```diff
@@ -105,7 +109,49 @@ Resources:
             Condition:
               StringNotEquals:
                 cloudformation:RoleARN: !Sub "arn:aws:iam::${AWS::AccountId}:role/admin/CloudFormationService"
-
+  Developer:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: Developer
+      Path: /admin/
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+        - Effect: Allow
+          Principal: {Federated: !Sub "arn:aws:iam::${AWS::AccountId}:saml-provider/Google"}
+          Action: sts:AssumeRoleWithSAML
+          Condition:
+            StringEquals:
+              SAML:aud: https://signin.aws.amazon.com/saml
+        - Effect: Allow
+          Principal: {Federated: accounts.google.com}
+          Action: sts:AssumeRoleWithWebIdentity
+          Condition:
+            StringEquals:
+              accounts.google.com:aud: '{{resolve:secretsmanager:admin/google_client_id}}'
+            ForAnyValue:StringEquals:
+              accounts.google.com:sub: !Ref GoogleDeveloperIDs
+        - Effect: Allow
+          Principal: {Service: ec2.amazonaws.com}
+          Action: sts:AssumeRole
+      Policies:
+      # Only allow developers to create/update stacks in specific environments.
+      - PolicyName: ProtectedStacks
+        PolicyDocument:
+          Statement:
+          - Effect: Deny
+            Action:
+            - cloudformation:CreateStack
+            - cloudformation:UpdateStack
+            - cloudformation:DeleteStack
+            - cloudformation:CreateChangeSet
```

Comment:
is CreateChangeSet needed to run the validate rake task? We still want developers to be able to validate, right?

Reply:
Good question- yes, `CreateChangeset` is needed to run `stack:validate` to show the resource-level diff on existing stacks, which I suppose we still _do_ want developers to be able to run on the CI-managed application stacks (e.g., to test out template changes in a PR before merging). I think changing the `CreateChangeSet` to `ExecuteChangeSet` will work for this.